rag/README.md

```diff
@@ -0,0 +1,100 @@
+# Retrieval Augmented Generation (RAG)
+
+This demo provides an example of using Retrieval Augmented Generation (RAG) to add additional context to an LLM chatbot. 
+
+## Build and Deploy Locally
+
+### Download LLM model(s)
+
+The two models that we have tested and recommend for this example are Llama2 and Mistral. The locations of the GGUF variants
+are listed below:
+
+* Llama2 - https://huggingface.co/TheBloke/Llama-2-7B-Chat-GGUF/tree/main
+* Mistral - https://huggingface.co/TheBloke/Mistral-7B-Instruct-v0.1-GGUF/tree/main
+
+_For a full list of supported model variants, please see the "Supported models" section of the
+[llama.cpp repository](https://github.com/ggerganov/llama.cpp?tab=readme-ov-file#description)._
+
+This example assumes that the developer already has a copy of the model that they would like to use downloaded onto their host machine and located in the `/models` directory of this repo. 
+
+This can be accomplished with:
+
+```bash
+cd models
+wget https://huggingface.co/TheBloke/Llama-2-7B-Chat-GGUF/resolve/main/llama-2-7b-chat.Q5_K_S.gguf
+cd ../
+```
+### Download the embedding model
+
+To encode our additional data and populate our vector database, we need an embedding model (a second language model) for this workflow. Here we will use `BAAI/bge-large-en-v1.5` all the necessary model files can be found and downloaded from https://huggingface.co/BAAI/bge-large-en-v1.5.
+
+
+Alternatively, you can run the below python code to download the model files directly into the `models/` directory.
+
+```python
+from langchain.embeddings.sentence_transformer import SentenceTransformerEmbeddings
+SentenceTransformerEmbeddings(model_name="BAAI/bge-base-en-v1.5",cache_folder="models/")
+```
+
+### Prepare the RAG dataset
+
+Once you have the embedding model in place, you will want to create a vector database with your custom data that can be used to augment our chatbot. The python code below will create a persistent vector database on our local machine that we can query at runtime. The code below simply uses the `fake_meeting.txt` demo file already included in this repository. Feel free to replace this with your own data. 
+
+```python
+from langchain_community.document_loaders import TextLoader
+from langchain.text_splitter import CharacterTextSplitter
+from langchain.embeddings.sentence_transformer import SentenceTransformerEmbeddings
+from langchain.vectorstores import Chroma
+
+raw_documents = TextLoader("data/fake_meeting.txt").load()
+text_splitter = CharacterTextSplitter(separator = ".", chunk_size=150, chunk_overlap=0)
+docs = text_splitter.split_documents(raw_documents)
+e = SentenceTransformerEmbeddings(model_name="BAAI/bge-base-en-v1.5",cache_folder="models/")
+db = Chroma.from_documents(docs,e,persist_directory="./data/chromaDB")
+```
```

Comment:
I copy/paste this script to run the sample but can we save it somewhere? So that it accept the document url to load.

Reply:
yeah, sure. I can work on a python script that can just do this for a user given the documents location. Let's do this in a follow up PR though if that is ok.

Reply:
Sure, ok for me 👍